features/enlinkd/persistence/api/src/main/java/org/opennms/netmgt/enlinkd/model/UserDefinedLink.java

```diff
@@ -0,0 +1,153 @@
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+
+@Entity
+@Table(name="user_defined_links")
+@XmlRootElement(name="user-defined-link")
+@XmlAccessorType(XmlAccessType.NONE)
+public class UserDefinedLink {
+
+    @XmlElement(name="node-id-a")
+    private Integer nodeIdA;
+    @XmlElement(name="component-label-a")
+    private String componentLabelA;
+    @XmlElement(name="node-id-z")
+    private Integer nodeIdZ;
+    @XmlElement(name="component-label-z")
+    private String componentLabelZ;
+    @XmlElement(name="link-id")
+    private String linkId;
+    @XmlElement(name="link-label")
+    private String linkLabel;
+    @XmlElement(name="owner")
+    private String owner;
+    @XmlElement(name="db-id")
+    private Integer dbId;
+
+    @Column(name="node_id_a", nullable = false)
+    public Integer getNodeIdA() {
+        return nodeIdA;
+    }
+
+    public void setNodeIdA(Integer nodeIdA) {
+        this.nodeIdA = nodeIdA;
+    }
+
+    @Column(name="component_label_a")
+    public String getComponentLabelA() {
+        return componentLabelA;
+    }
+
+    public void setComponentLabelA(String componentLabelA) {
+        this.componentLabelA = componentLabelA;
+    }
+
+    @Column(name="node_id_z", nullable = false)
+    public Integer getNodeIdZ() {
+        return nodeIdZ;
+    }
+
+    public void setNodeIdZ(Integer nodeIdZ) {
+        this.nodeIdZ = nodeIdZ;
+    }
+
+    @Column(name="component_label_z")
+    public String getComponentLabelZ() {
+        return componentLabelZ;
+    }
+
+    public void setComponentLabelZ(String componentLabelZ) {
+        this.componentLabelZ = componentLabelZ;
+    }
+
+    @Column(name="link_id", nullable = false)
+    public String getLinkId() {
+        return linkId;
+    }
+
+    public void setLinkId(String linkId) {
+        this.linkId = linkId;
+    }
+
+    @Column(name="link_label")
+    public String getLinkLabel() {
+        return linkLabel;
+    }
+
+    public void setLinkLabel(String linkLabel) {
+        this.linkLabel = linkLabel;
+    }
+
+    @Column(name="owner", nullable = false)
+    public String getOwner() {
+        return owner;
+    }
+
+    public void setOwner(String owner) {
+        this.owner = owner;
+    }
+
+    @Id
+    @Column(name="id", nullable = false)
+    @SequenceGenerator(name = "opennmsSequence", sequenceName = "opennmsNxtId")
+    @GeneratedValue(generator = "opennmsSequence")
+    public Integer getDbId() {
+        return dbId;
+    }
+
+    public void setDbId(Integer dbId) {
+        this.dbId = dbId;
+    }
+
+    @Override
+    public String toString() {
```

Comment:
I have had pull request comments asking to use the Guava Objects.toStringHelper for toString() method implementations. Not sure if that is a rule we are trying to follow?

    @Override
    public String toString() {
        return MoreObjects.toStringHelper(this)
            .add("nodeIdA=", nodeIdA)
            .add("nodeIdZ=", nodeIdZ)
// ...
            .toString();
    }